### PR TITLE
[W4.4b][F12-A1]Seunghwan Choi

### PR DESCRIPTION
**Pull request by shadow2496**
Implement a class [LO-ImplementClass]

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -21,36 +35,72 @@
...
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        if (!isValidAddress(address.trim())) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        this.block = new Block(matcher.group("block"));
+        this.street = new Street(matcher.group("street"));
+        this.unit = new Unit(matcher.group("unit"));
+        this.postalCode = new PostalCode(matcher.group("postalCode"));
```
Comment (erik0704) [inline src/seedu/addressbook/data/person/Address.java:45]: Avoid unnecessary use of `this` with fields: Use the `this` keyword only when a field is shadowed by a method or constructor parameter.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -21,36 +35,72 @@
...
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        if (!isValidAddress(address.trim())) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        this.block = new Block(matcher.group("block"));
```
Comment (erik0704) [inline src/seedu/addressbook/data/person/Address.java:42]: Good job user `matcher` here.

**Review by erik0704 — COMMENTED** [review]
@shadow2496 Some comments added. Please close the PR after reading comments.

File: src/seedu/addressbook/data/person/Address.java (hunk 1 above)
Comment (shadow2496) [reply src/seedu/addressbook/data/person/Address.java:45]: I got it now, thank you.

**Pull request closed without merging**